tests/TurboMqtt.Tests/PacketGenerators.cs

```diff
@@ -0,0 +1,117 @@
+            MqttPacketType.ConnAck,
+            MqttPacketType.Publish,
+            MqttPacketType.PubAck,
+            MqttPacketType.PubRec,
+            MqttPacketType.PubRel,
+            MqttPacketType.PubComp,
+            MqttPacketType.Subscribe,
+            MqttPacketType.SubAck,
+            MqttPacketType.Unsubscribe,
+            MqttPacketType.UnsubAck,
+            MqttPacketType.PingReq,
+            MqttPacketType.PingResp,
+            MqttPacketType.Disconnect
+        ]);
+
+        return Arb.From(g);
+    }
+
+    public static Arbitrary<MqttPacket> ConnectPacketArb()
+    {
+        return (from protocolVersion in Gen.Constant(MqttProtocolVersion.V3_1_1)
+            from clientId in
+                Arb.Generate<string>()
+                    .Where(s => !string.IsNullOrWhiteSpace(s) && s.Length > 0 && MqttClientIdValidator.ValidateClientId(s).IsValid) // ensure clientId is not null or whitespace
+            from cleanSession in Arb.Generate<bool>()
+            from keepAlive in Arb.Generate<ushort>()
+            select (MqttPacket)new ConnectPacket(protocolVersion)
+            {
+                ClientId = clientId,
+                ConnectFlags = new ConnectFlags()
+                {
+                    CleanSession = cleanSession
+                },
+                KeepAliveSeconds = keepAlive
+            }).ToArbitrary();
+    }
+
+    public static Arbitrary<MqttPacket> PublishPacketArb()
+    {
+        return (from qos in Arb.Generate<QualityOfService>()
+            from duplicate in Arb.Generate<bool>()
+            from retainRequested in Arb.Generate<bool>()
+            from topicName in
+                Arb.Generate<string>()
+                    .Where(s => !string.IsNullOrWhiteSpace(s) && s.Length > 0 && MqttTopicValidator.ValidatePublishTopic(s).IsValid) // ensure topicName is not null or whitespace
+            from payloadLength in Gen.Choose(0, 16 * 1024) // You can adjust the max size as needed
+            from bytes in Gen.ArrayOf(payloadLength, Arb.Generate<byte>())
+            from packetId in Arb.Generate<ushort>()
+            select (MqttPacket)new PublishPacket(qos, duplicate, retainRequested, topicName)
+            {
+                Payload = new ReadOnlyMemory<byte>(bytes),
+                PacketId = packetId
+            }).ToArbitrary();
+    }
+    
+    public static Arbitrary<MqttPacket> PacketArb()
+    {
+        return Gen.OneOf(
+            ConnectPacketArb().Generator,
+            PublishPacketArb().Generator
+        ).ToArbitrary();
+    }
+    
+    public static Arbitrary<ReadOnlyMemory<byte>[]> FragmentedPackets(Arbitrary<MqttPacket> packetArb)
+    {
+        // need help here
+        var serializedPackets = packetArb.Generator.Select(packet =>
+        {
+            var estimatedSize = MqttPacketSizeEstimator.EstimateMqtt3PacketSize(packet);
+            var fullSize = MqttPacketSizeEstimator.GetPacketLengthHeaderSize(estimatedSize) + estimatedSize + 1;
```

Comment:
Added a +1 here. `GetPacketLengthHeaderSize` only calculates the payload length, still need to add 1 for the control byte

Reply:
LGTM